index.html

```diff
@@ -5,45 +5,364 @@
+      data.sort(function (a, b) { return b.avg_family_income - a.avg_family_income; });
+    } else if (options.sorted === 'desc-poverty') {
+      data.sort(function (a, b) { return b.poverty_rate - a.poverty_rate; });
+    }
+ }
+
+//My attempt at Pagination for Next button, not working though
+
+
+
+  if (options.page) {
+    if (options.page === 'next-page'){
+      
+      data = data.slice(function (pageno, max_num) {
+        console.log('3');
+        var pageno = 2;
+        var rows = 10;
+        
+        return data.slice((pageno - 1) * rows, pageno * rows); });
+    }
+  } 
+
+  var table = document.getElementById('data-table');
+  var tbody = table.getElementsByTagName('tbody')[0];
+
+  tbody.innerHTML = '';
+
+  data.forEach(function (d) {
+
+    var row = document.createElement('tr');
+
+    var name = document.createElement('td');
+    name.innerHTML = d.neighborhood;
+    row.appendChild(name);
+
+    var income= document.createElement('td');
+    income.innerHTML = '$' + new Intl.NumberFormat().format(d.avg_family_income);
+    row.appendChild(income);
+
+    var barWrapper = document.createElement('td');
+    var bar = document.createElement('div');
+    bar.setAttribute('class', 'bar');
+    bar.setAttribute('style', 'width: ' + (d.avg_family_income / MAX_INCOME * 100) + '%;');
+    barWrapper.appendChild(bar);
+    row.appendChild(barWrapper);
+
+    var poverty= document.createElement('td');
+    poverty.innerHTML = d.poverty_rate + '%';
+    row.appendChild(poverty);
+
+    var barWrapper = document.createElement('td');
+    var bar = document.createElement('div');
+    bar.setAttribute('class', 'bar');
+    bar.setAttribute('style', 'width: ' + (d.poverty_rate / MAX_POVERTY * 100) + '%;');
+    barWrapper.appendChild(bar);
+    row.appendChild(barWrapper);
+
+
+
+    tbody.appendChild(row);
+  });
+}
+
+updateTable();
+
+  
+
+
+document.getElementById('sort-desc-income').addEventListener('click', function() {
+  if (options.sorted === 'desc') {
+    options.sorted = false;
+  } else {
+    options.sorted = 'desc';
+  }
+  updateTable();
+});
+
+document.getElementById('sort-desc-poverty').addEventListener('click', function() {
+  if (options.sorted === 'desc-poverty') {
+    options.sorted = false;
+  } else {
+    options.sorted = 'desc-poverty';
+  }
+  updateTable();
+});
+
+document.getElementById('filter-poverty').addEventListener('click', function() {
+  if (options.filtered === 'filter-p') {
+    options.filtered = false;
+  } else {
+    options.filtered = 'filter-p';
+  }
+  updateTable();
+});
+
+document.getElementById('filter-income').addEventListener('click', function() {
+  if (options.filtered === 'filter-i') {
+    options.filtered = false;
+  } else {
+    options.filtered = 'filter-i';
+  }
+  updateTable();
+});
+
+document.getElementById('page-next').addEventListener('click', function() {
+  if (options.page === 'next-page') {
+    options.page = 0;
+  } else {
+    options.page = 'next-page';
```

Comment:
You're very much on the right track here. I really like that you're keeping this event handler simple, and putting the more complex logic in `updateTable()`. But consider whether you could do this by, instead of adding a "next-page" flag that you'll have to deal with elsewhere, you just set options.page to 1 (or for more resilient code, `options.page = options.page + 1;`).